[DO NOT MERGE] Add torch-2.8 variant config for XPU

dvrogozh opened the pull request:
Submitting as draft as torch-2.8 is already done and it's unlikely we will merge this PR as is. More likely we will need to prepare another PR for upcoming torch-2.9.

Tried with:
```
wget https://download.pytorch.org/whl/xpu/torch-2.8.0%2Bxpu-cp312-cp312-linux_x86_64.whl

variant_repack build -i $(ls *xpu*.whl) -o ./torch/ \
  --pyproject-toml ./torch_pyproject.toml \
  --variant-config-toml ./torch_variant_config.toml \
  --variant-config-name xpu \
  --metadata-config-name torch
```

Dependencies taken unchanged from https://download.pytorch.org/whl/xpu.

CC: @atalman, @chuanqi129

dvrogozh commented:
These are steps I used to verify pytorch xpu variant build (these are steps, not a working shell script as there are few missing parts which are assumed such as installed or built variantlib, variant-repack and intel-variant-provider):
```
LOCAL_INDEX=~/tmp/repack/xpu/
CONFIG=~/git/variant-repack/configs/torch-2.8/

mkdir $LOCAL_INDEX

# Download XPU wheels with dependencies:
# torchvision will download torch as a dependency, we need both
python3 -m pip download torchvision --index-url https://download.pytorch.org/whl/xpu --dest $LOCAL_INDEX

# Create directory structure to serve as --index-url
cd $LOCAL_INDEX
for pkg in `ls *.whl`; do name=${pkg%%-*}; name=${name//_/-}; mkdir $name && mv $pkg $name/; done
mv Jinja2/ jinja2
mv MarkupSafe/ markupsafe

# Move original torch away and repack with variants:
mv torch torch-no-variant && mkdir torch
mv torchvision torchvision-no-variant && mkdir torchvision

variant_repack build -i $(ls torch-no-variant/*xpu*.whl) -o ./torch/ --pyproject-toml $CONFIG/torch_pyproject.toml --variant-config-toml $CONFIG/torch_variant_config.toml --variant-config-name intel --metadata-config-name torch

variant_repack build -i $(ls torchvision-no-variant/*xpu*.whl) -o ./torchvision/ --pyproject-toml $CONFIG/torch_pyproject.toml --variant-config-toml $CONFIG/torch_variant_config.toml --variant-config-name intel --metadata-config-name torchvision

# That’s a better variant as it checks some errors in package configurations:
variantlib generate-index-json -d torch/
variantlib generate-index-json -d torchvision/
# or
cp $CONFIG/torch-2.8.0-variants.json torch/
cp $CONFIG/torchvision-2.8.0-variants.json torch/

# Now we can server the index:
# these are previously built with python3 -m build -w .
mkdir nvidia-variant-provider/ && cp ~/git/nvidia-variant-provider/dist/*.whl nvidia-variant-provider/
mkdir intel-variant-provider/ && cp ~/git/intel-variant-provider/dist/*.whl intel-variant-provider/

python -m http.server 8000
```
And in other shell we can install (on Intel GPU compatible system this should install xpu pytorch variant):
```
python3 -m pip install --verbose --index-url http://localhost:8000/ torchvision
```

dvrogozh commented:
Modified `variants.json` file can be found in the PR.

Here is some info on the produced package:
```
$ variantlib analyze-wheel -i torch/torch-2.8.0-cp312-cp312-linux_x86_64-xpu.whl
variantlib.commands.analyze_wheel - INFO - Filepath: `torch-2.8.0-cp312-cp312-linux_x86_64-xpu.whl` ... is a Wheel Variant - Label: `xpu`
############################## Variant: `dceff3be` #############################
intel :: device_ip :: 12.55.8
intel :: device_ip :: 12.60.7
intel :: device_ip :: 12.71.4
intel :: device_ip :: 12.74.4
intel :: device_ip :: 20.1.0
intel :: device_ip :: 20.4.4
################################################################################

$ variantlib analyze-wheel -i torchvision/torchvision-0.23.0-cp312-cp312-manylinux_2_28_x86_64-xpu.whl
variantlib.commands.analyze_wheel - INFO - Filepath: `torchvision-0.23.0-cp312-cp312-manylinux_2_28_x86_64-xpu.whl` ... is a Wheel Variant - Label: `xpu`
############################## Variant: `dceff3be` #############################
intel :: device_ip :: 12.55.8
intel :: device_ip :: 12.60.7
intel :: device_ip :: 12.71.4
intel :: device_ip :: 12.74.4
intel :: device_ip :: 20.1.0
intel :: device_ip :: 20.4.4
################################################################################
```

Here are all `torch` XPU wheel dependencies for pytorch 2.8 (as also can be seen in METADATA file below):
```
Metadata-Version: 2.4
Name: torch
Version: 2.8.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Download-URL: https://github.com/pytorch/pytorch/tags
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3-Clause
Project-URL: Homepage, https://pytorch.org/
Project-URL: Documentation, https://pytorch.org/docs/
Project-URL: Source, https://github.com/pytorch/pytorch
Project-URL: Forum, https://discuss.pytorch.org/
Keywords: pytorch,machine learning
Classifier: Development Status :: 5 - Production/Stable
Classifier: Intended Audience :: Developers
Classifier: Intended Audience :: Education
Classifier: Intended Audience :: Science/Research
Classifier: License :: OSI Approved :: BSD License
Classifier: Topic :: Scientific/Engineering
Classifier: Topic :: Scientific/Engineering :: Mathematics
Classifier: Topic :: Scientific/Engineering :: Artificial Intelligence
Classifier: Topic :: Software Development
Classifier: Topic :: Software Development :: Libraries
Classifier: Topic :: Software Development :: Libraries :: Python Modules
Classifier: Programming Language :: C++
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.9
Classifier: Programming Language :: Python :: 3.10
Classifier: Programming Language :: Python :: 3.11
Classifier: Programming Language :: Python :: 3.12
Classifier: Programming Language :: Python :: 3.13
Requires-Python: >=3.9.0
Description-Content-Type: text/markdown
License-File: LICENSE
License-File: NOTICE
Provides-Extra: optree
Provides-Extra: opt-einsum
Provides-Extra: pyyaml
Dynamic: author
Dynamic: author-email
Dynamic: classifier
Dynamic: description
Dynamic: description-content-type
Dynamic: download-url
Dynamic: home-page
Dynamic: keywords
Dynamic: license-file
Dynamic: provides-extra
Dynamic: requires-dist
Dynamic: requires-python
Dynamic: summary
Requires-Dist: dpcpp-cpp-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: filelock
Requires-Dist: fsspec
Requires-Dist: impi-rt==2021.15.0; platform_system == "Linux" and platform_machine == "x86_64" and "intel" in variant_namespaces
Requires-Dist: intel-cmplr-lib-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-cmplr-lib-ur==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-cmplr-lic-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-opencl-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-openmp==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-pti==0.12.3; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: intel-sycl-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: jinja2
Requires-Dist: mkl==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: networkx
Requires-Dist: nvidia-cublas==12.6.4.1; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cublas==12.8.4.1; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cublas==12.9.1.4; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cuda-cupti==12.6.80; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-cupti==12.8.90; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-cupti==12.9.79; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cuda-nvrtc==12.6.77; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-nvrtc==12.8.93; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-nvrtc==12.9.86; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cuda-runtime==12.6.77; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-runtime==12.8.90; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cuda-runtime==12.9.79; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cudnn==9.10.2.21; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia" in variant_namespaces
Requires-Dist: nvidia-cufft==11.3.0.4; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cufft==11.3.3.83; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cufft==11.4.1.4; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cufile==1.11.1.6; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cufile==1.13.1.3; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cufile==1.14.1.1; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-curand==10.3.10.19; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-curand==10.3.7.77; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-curand==10.3.9.90; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cusolver==11.7.1.2; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cusolver==11.7.3.90; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cusolver==11.7.5.82; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cusparse==12.5.10.65; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-cusparse==12.5.4.2; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cusparse==12.5.8.93; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-cusparselt==0.7.1; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia" in variant_namespaces
Requires-Dist: nvidia-nccl==2.27.3; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia" in variant_namespaces
Requires-Dist: nvidia-nvjitlink==12.6.85; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-nvjitlink==12.8.93; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-nvjitlink==12.9.86; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: nvidia-nvtx==12.6.77; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-nvtx==12.8.90; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" not in variant_properties
Requires-Dist: nvidia-nvtx==12.9.79; platform_system == "Linux" and platform_machine == "x86_64" and "nvidia :: cuda_version_lower_bound :: 12.0" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.8" not in variant_properties and "nvidia :: cuda_version_lower_bound :: 12.9" in variant_properties
Requires-Dist: oneccl-devel==2021.15.2; platform_system == "Linux" and platform_machine == "x86_64" and "intel" in variant_namespaces
Requires-Dist: oneccl==2021.15.2; platform_system == "Linux" and platform_machine == "x86_64" and "intel" in variant_namespaces
Requires-Dist: onemkl-sycl-blas==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: onemkl-sycl-dft==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: onemkl-sycl-lapack==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: onemkl-sycl-rng==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: onemkl-sycl-sparse==2025.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: opt-einsum>=3.3; extra == "opt-einsum"
Requires-Dist: optree>=0.13.0; extra == "optree"
Requires-Dist: pytorch-triton-xpu==3.4.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: pyyaml; extra == "pyyaml"
Requires-Dist: setuptools; python_version >= "3.12"
Requires-Dist: sympy>=1.13.3
Requires-Dist: tbb==2022.1.0; platform_system == "Linux" and "intel" in variant_namespaces
Requires-Dist: tcmlib==1.3.0
Requires-Dist: triton==3.4.0; platform_system == "Linux" and "nvidia" in variant_namespaces
Requires-Dist: typing-extensions>=4.10.0
Requires-Dist: umf==0.10.0; platform_system == "Linux" and "intel" in variant_namespaces
```

Here is beginning of the METADATA file of the installed `torchvision` package (I've cut .md readme part):
```
Metadata-Version: 2.1
Name: torchvision
Version: 0.23.0
Summary: image and video datasets and models for torch deep learning
Home-page: https://github.com/pytorch/vision
Author: PyTorch Core Team
Author-email: soumith@pytorch.org
License: BSD
Requires-Python: >=3.9
Description-Content-Type: text/markdown
License-File: LICENSE
Provides-Extra: gdown
Provides-Extra: scipy
Requires-Dist: gdown (>=4.7.3) ; extra == 'gdown'
Requires-Dist: numpy
Requires-Dist: pillow (!=8.3.*,>=5.3.0)
Requires-Dist: scipy ; extra == 'scipy'
Requires-Dist: torch==2.8.0
```

dvrogozh commented:
@atalman, can you, please, help to clarify why pytorch `variant-repack` modifies dependencies list of non-variant pytorch wheels? I post some of my comments below why I don't see that as required.

Per my understanding variants are installed as follows:
1. pip looks for `variants.json` file and installs variant providers
2. Variant providers run one by one and return environment description
3. Pytorch build variant is selected for installation
4. Build variant is getting installed

Pytorch has few separate variant .whl files, few for cuda, one for xpu, etc. Each of these wheel files already has unique set of dependencies pinned for specific versions. Even for different CUDA variants pinned versions are different. Thus, it seems that original non-variant .whl file dependencies would be left unmodified, then all still would work as selected variant would just install its specific set of dependencies.

dvrogozh commented:
@atalman : can you, please, hold off merging for 1 day as we consider to rename xpu variant plugin to intel variant plugin?

DEKHTIARJonathan commented:
We need to find a way for all wheels to include all dependencies for all variants.
Which means we need to find a way to re-unify all the dependencies and gate them by variant selectors

dvrogozh commented:
> We need to find a way for all wheels to include all dependencies for all variants.

@DEKHTIARJonathan, can you, please, clarify why do we need to do that? Is pip install doing something else than what I described in https://github.com/wheelnext/variant-repack/pull/3#issuecomment-3198845620?

DEKHTIARJonathan commented:
Sure - there's a few answers to that.

Up until today we didn't have to worry about this because wheels were hosted on different indexes.
Well now it's different. Technically `pip` wouldn't care, so we don't have to worry. However, it's pretty much the only installer that doesn't `uv`, `pdm`, `poetry` all approach installation as universal resolution and assume all wheels share the same dependencies (gated with selectors).

Unfortunately that behavior is not a "hard requirement" of the ecosystem. But a behavior adopted by 99.999...% of wheels.
Astral team has been very insisting that we should design `variant` to respect that design principle, and I agree with them here.

Now what does it mean for us, we need adjust `variant-repack` to allow that behavior. It's not a big deal, but we need to implement it :)

dvrogozh commented:
> @atalman : can you, please, hold off merging for 1 day as we consider to rename xpu variant plugin to intel variant plugin?

I've updated the PR with the renamed xpu variant plugin.

At the moment I did not add XPU dependencies to "all the wheels". @atalman, @DEKHTIARJonathan, is that something I need to do before merging this PR or you want to do that on your own after the PR merge?

DEKHTIARJonathan commented:
No worries there's zero rush we have 2 months to "figure things out" :) 
Take your time.

And we still need to find a good idea to merge dependencies.

dvrogozh commented:
@atalman : I pushed a pre-release tag for intel variant provider. You can find plugin .whl build in there:
* https://github.com/wheelnext/intel-variant-provider/releases/tag/v0.0.1-rc1

atalman commented:
HI @DEKHTIARJonathan I don't think we have 2 month time. According to dev discuss: https://dev-discuss.pytorch.org/t/reminder-calls-for-features-upcoming-branch-cut/3225

Release date: ``M6: Release Day (10/15)``

dvrogozh commented:
@atalman : I've just rebased this PR and added a commit which adds intel deps to the common list of deps for all wheels (intel deps are protected by namespace check similar to nvidia ones). I still think we can do simpler repacking by automatically adding namespace check for the variant dependencies and doing more sophisticated stuff if needed in the config file. However, this optimization can be done next time. For now I follow current design.

dvrogozh commented on configs/torch-2.8/torch_variant_config.toml:
```diff
@@ -64,6 +86,28 @@ deps_add_list    = [
+
+    # XPU
+    'dpcpp-cpp-rt==2025.1.1; platform_system == "Linux" and "intel" in variant_namespaces',
+    'impi-rt==2021.15.0; platform_system == "Linux" and platform_machine == "x86_64" and "intel" in variant_namespaces',
```
If anyone curious, few lines have additional `platform_machine == "x86_64"` because that's how original xpu non-variant wheel has specified these dependencies. See here: https://github.com/wheelnext/variant-repack/pull/3#issuecomment-3198830400. I basically copied xpu specific deps here and added 2 things: namespace protection (`"intel" in variant_namespaces'`) and affinity to Linux (`platform_system == "Linux"`).

jithunnair-amd commented:
@dvrogozh Shouldn't the version say just `0.23.0` for a variant wheel?

> Here is beginning of the METADATA file of the installed torchvision package (I've cut .md readme part):
```
Metadata-Version: 2.1
Name: torchvision
Version: 0.23.0+xpu
```

dvrogozh commented:
> @dvrogozh Shouldn't the version say just 0.23.0 for a variant wheel?

I think I copied from non-variant installation. The package metadata for torch also got changed after https://github.com/wheelnext/variant-repack/pull/3#issuecomment-3250543438, so I've updated a comment above to match metadata as it currently appears on the variant installation.

DEKHTIARJonathan commented:
Closing this PR is not needed
